src/encoded/static/components/rnaget.js

```diff
@@ -657,42 +694,81 @@ class RNASeqMatrixSearch extends TextFilter {
     constructor(props) {
         super();
 
-        this.handleChange = this.handleChange.bind(this);
+        this.onUnitsChange = this.onUnitsChange.bind(this);
+        this.onKeyDown = this.onKeyDown.bind(this);
         this.handleInputChange = this.handleInputChange.bind(this);
         this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleAutocompleteClick = this.handleAutocompleteClick.bind(this);
+        this.lastGene = this.lastGene.bind(this);
 
         this.state = {
             unitsOption: props.query.getKeyValuesIfPresent('units').join(','),
             genes: props.query.getKeyValuesIfPresent('genes').join(','),
+            showAutoSuggest: false,
         };
     }
 
-    handleChange(e) {
+    onUnitsChange(e) {
         this.setState({ unitsOption: e.target.value });
     }
 
-    handleInputChange(e) {
-        this.setState({ genes: e.target.value });
+    lastGene() {
+        const inputNode = this.geneInput;
+
+        if (inputNode === undefined) {
```

Comment:
Why look for undefined exactly and not falsey?

Reply:
falsey is indeed better, pushing a change...